### ISSUE TITLE
r/aws_vpn_gateway_attachment: Incorrect error code for missing VPN gateway

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#4894]
When deleting an `aws_vpn_gateway_attachment`:
```
* aws_vpn_gateway_attachment.test2: Error detaching VPN Gateway "vgw-xxxxxxxx" from VPC "vpc-xxxxxxxx": InvalidVpnGatewayID.NotFound: The vpnGateway ID 'vgw-xxxxxxxx' does not exist
	status code: 400, request id: 15974695-8ffc-45e0-82c0-21d3df0d2c49
```

The VPN gateway attachment code is using an incorrectly capitalized AWS error code for the missing VPN gateway: https://github.com/terraform-providers/terraform-provider-aws/blob/b040e1de45f51432347bef28f8c4dccaeec425c5/aws/resource_aws_vpn_gateway_attachment.go#L133

The correctly capitalized AWS error code is used in the `aws_vpn_gateway` resource: https://github.com/terraform-providers/terraform-provider-aws/blob/b040e1de45f51432347bef28f8c4dccaeec425c5/aws/resource_aws_vpn_gateway.go#L171

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.24.0 of the AWS provider, very shortly.

**Comment by bflad** [comment]
This has been released in [version 1.24.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1240-june-21-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
